Serverless changelog for April 20, 2026 (#6039)
elastic/docs-content: release-notes/elastic-cloud-serverless/index.md

* Fixes the tooltip position for the help button on dashboard Markdown editors so it no longer blocks the preview toggle [#261596]({{kib-pull}}261596)
* Fixes a bug where saving a visualization from an AI Agent chat to the current dashboard required a manual page refresh [#262107]({{kib-pull}}262107)
* Clears `location.state` after a locator merge to fix stale reload overrides [#262695]({{kib-pull}}262695)
* Fixes permissions for span events stored in logs data streams [#263415]({{kib-pull}}263415)
* Disables the output selector for managed policies in the package policy edit form [#263494]({{kib-pull}}263494)
* Fixes package policy count filters to use `NOT latest_revision:false` instead of `latest_revision:true` [#263717]({{kib-pull}}263717)
* Shows all `source_index` entries when `source_index` is an array [#261875]({{kib-pull}}261875)
* Resets the time field when the updated index pattern does not have it [#262001]({{kib-pull}}262001)
* Reports server-side control errors correctly [#263020]({{kib-pull}}263020)
* Fixes the tab URL state leaking when leaving Discover [#262929]({{kib-pull}}262929)
* Handles Dagre layout failures on the service map [#262240]({{kib-pull}}262240)

> * Fixes APM routing and settings [#262723]({{kib-pull}}262723)

Reviewer: ```suggestion
* Fixes APM rule charts and selectors to display data from linked projects when using cross-project search [#262723]({{kib-pull}}262723)
```

* Implements a self-healing mechanism for malformed URLs to prevent plugin crashes [#257245]({{kib-pull}}257245)

> * Aligns the serverless APM service navigation with the stateful version [#263119]({{kib-pull}}263119)

Reviewer: ```suggestion
* Aligns the Serverless APM service navigation with the stateful version [#263119]({{kib-pull}}263119)
```

* Updates the **Failed Tests by Step** panel to respect time range changes in Synthetics [#263317]({{kib-pull}}263317)
* Fixes the Elastic Inference Service callout layout [#262780]({{kib-pull}}262780)
* Casts dimensions to prevent conflict errors [#263472]({{kib-pull}}263472)
* Improves accessibility in the Discover UI [#262975]({{kib-pull}}262975)
* Fixes tooltip positioning in Lens [#262682]({{kib-pull}}262682)
* Re-validates the EQL query when the index pattern changes, clearing errors when the query is valid for the newly selected data view [#261027]({{kib-pull}}261027)
* Extends the query execution timeout to 24 hours [#262008]({{kib-pull}}262008)
* Restores the CrowdStrike `runscript` output display in the Response Console and Action History views [#262470]({{kib-pull}}262470)
* Fixes a whitespace bug in alert KPIs [#260803]({{kib-pull}}260803)
* Updates the text in the announcement modal [#263034]({{kib-pull}}263034)
* Classifies Detection Engine gap errors as user errors again [#263244]({{kib-pull}}263244)

> * Suppresses the announcement modal in automated browsers via `navigator.webdriver` [#263785]({{kib-pull}}263785)

Reviewer 2: ```suggestion
* Suppresses the announcement modal in automated browsers by checking `navigator.webdriver` before rendering [#263785]({{kib-pull}}263785)
```
I pulled the text from the associated PR.

* Opens the Agent Builder sidebar when navigating to rule creation [#262588]({{kib-pull}}262588)
* Changes the last event ingested logic to pull from more indices and the correct space [#262346]({{kib-pull}}262346)

> * Fixes incorrect capitalization of Elastic Inference and other menu labels in the serverless project sidebar [#263176]({{kib-pull}}263176)

Reviewer 2: ```suggestion
* Fixes incorrect capitalization of Elastic Inference and other menu labels in the Serverless project sidebar [#263176]({{kib-pull}}263176)
```

> * Shows a save confirmation message after saving changes on the **Feature Settings** page [#263768]({{kib-pull}}263768)

Reviewer: Maybe this is redundant?

```suggestion
* Shows a confirmation message after saving changes on the **Feature Settings** page [#263768]({{kib-pull}}263768)
```

* Ensures the single metric chart in Anomaly Explorer shows anomaly actions correctly [#263925]({{kib-pull}}263925)
* Fixes a bug where the agent ID was incorrectly reset [#263044]({{kib-pull}}263044)
* Fixes the SML index type mismatch [#263754]({{kib-pull}}263754)
* Upgrades the `hono` and `@hono/node-server` dependencies [#263794]({{kib-pull}}263794)
* Fixes the `offHeap` size for empty indices in DiskBBQ [#146347]({{es-pull}}146347)
* Refactors the Contextual AI integration in the Inference API and fixes multiple issues [#145700]({{es-pull}}145700)
* Omits uncomputed model stats in {{ml}} [#146186]({{es-pull}}146186)
* Fixes {{esql}} `TBUCKET(n)` on an empty filter range [#146366]({{es-pull}}146366)
* Fixes {{esql}} `starts_with`/`ends_with` with special characters [#146348]({{es-pull}}146348)
* Fixes heap attack vulnerabilities in the {{esql}} `SPARKLINE` function [#146503]({{es-pull}}146503)
* Fixes single file splitting and bzip2 bytes counting in {{esql}} external sources [#146373]({{es-pull}}146373)
* Ensures the {{esql}} NDJSON plugin respects the error policy [#146433]({{es-pull}}146433)
* Fixes bugs in the {{esql}} NDJSON plugin [#146491]({{es-pull}}146491)

> * Fixes the `deriv` aggregation function intermediate state [#146160]({{es-pull}}146160)

Reviewer: ```suggestion
* Fixes the `deriv` aggregation function in {{esql}} to correctly handle intermediate states [#146160]({{es-pull}}146160)
```
